PennMobile/GSR-Booking/Controllers/GSRController.swift

```diff
@@ -134,24 +145,31 @@ extension GSRController: GSRViewModelDelegate {
         let date = viewModel.getSelectedDate()
         
         self.startLoadingViewAnimation()
-        
+                        
         GSRNetworkManager.instance.getAvailability(for: location.lid, date: date) { (rooms) in
             
             DispatchQueue.main.async {
-                if let rooms = rooms {
+                if let rooms = rooms {                    
                     self.viewModel.updateData(with: rooms)
                     self.refreshDataUI()
                     self.rangeSlider.reload()
                     self.refreshBarButton()
                     self.stopLoadingViewAnimation()
+                }else{//no valid response
+                    self.viewModel.updateData(with: [])
+                    self.refreshDataUI()
+                    self.rangeSlider.reload()
+                    self.refreshBarButton()
+                    self.stopLoadingViewAnimation()
```

Comment:
Suggestion for readability: You can remove the `if ... else ...` and instead write `let rooms = rooms ?? []` at the start. This removes the duplicate code.